lib/src/connector/connector.cc

```diff
@@ -259,37 +269,50 @@ void Connector::connect(int max_connect_attempts)
                         || session_association_.got_messaging_failure.load();
             });
 
+        // Look for failures
         if (session_association_.got_messaging_failure.load()) {
+            association_timings_.setCompleted(false);
             LOG_DEBUG("It seems that an error occurred during the Session "
-                      "Association ({1})",
+                      "Association ({1}) - {2}",
                       (session_association_.error.empty()
                             ? lth_loc::translate("undetermined error")
-                            : session_association_.error));
+                            : session_association_.error),
+                       association_timings_.toString());
             session_association_.reset();
             throw connection_association_error {
                 lth_loc::translate("invalid Associate Session response") };
         }
         if (session_association_.in_progress.load()) {
+            // HERE(ale): don't set the associate_timings_ completion
+            // as we can't be sure whether the request was sent
             LOG_DEBUG("Associate Session timed out");
             session_association_.reset();
             throw connection_association_error {
                 lth_loc::translate("operation timeout") };
         }
         if (!session_association_.success.load()) {
+            association_timings_.setCompleted(false);
+            LOG_DEBUG(association_timings_.toString());
             std::string msg { lth_loc::translate("Associate Session failure") };
-            LOG_DEBUG(msg);
             if (!session_association_.error.empty())
                 msg += ": " + session_association_.error;
             session_association_.reset();
             throw connection_association_response_failure { msg };
         }
+
+        // Success!
+        association_timings_.setCompleted();
+        LOG_DEBUG(association_timings_.toString(false));
     } catch (const connection_processing_error& e) {
         // NB: connection_fatal_errors (can't connect after n tries)
         //     and _config_errors (TLS initialization error) are
         //     propagated whereas _processing_errors must be converted
         //     to _config_errors (they can be thrown after the
         //     synchronous call websocketpp::Endpoint::connect())
-        LOG_DEBUG("Failed to establish the WebSocket connection: {1}", e.what());
+        LOG_DEBUG("Failed to establish the WebSocket connection ({1}) - ", e.what());
```

Comment:
Mistake. I was considering adding the timings everywhere, but we don't want them here, as the Association may not have started.